java-manual/modules/ROOT/pages/value-mapping.adoc

```diff
@@ -0,0 +1,173 @@
+This will make the mapped object available out of the scope of the method in which it was defined.
+
+[NOTE]
+====
+While constructor arguments with specific complex types (ex. `record Friends(List<String> names) {}`) are supported, constructor arguments with generic complex types (ex. `record Friends<T>(List<T> names) {}`) are not supported.
+====
+
+
+== Map properties with different names (`@Property`)
+
+A record's property names and its query return keys can be different.
+For example, consider a node `(:Person {name: "Alice"})`.
+The returned keys for the query `MERGE (p:Person {name: "Alice"}) RETURN p.name` are `p.name`, even if the property name is `name`.
+Similarly, for the query `MERGE (pers:Person {name: "Alice"}) RETURN pers.name`, the return keys are `pers.name`.
+
+You can always alter the return key with the Cypher operator link:https://neo4j.com/docs/cypher-manual/current/clauses/return/#return-column-alias[`AS`] (ex. `MERGE (p:Person {name: "Alice"}) RETURN p.name AS name`), or use the `@Property(<dbPropertyName>)` decorator to specify the property name that the following constructor argument should map to.
+
+.Map properties `name`/`age` to the object attributes `firstName`/`Years`
+[source, java]
+----
+// import org.neo4j.driver.mapping.Property;
+
+record Person(@Property("name") String firstName, @Property("age") Integer Years) {}
+var persons = driver.executableQuery("MERGE (p:Person {name: 'Margarida', age: 29}) RETURN p")
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.get("p").as(Person.class))
+    .toList();
+System.out.println(persons.get(0));  // Person[firstName=Margarida, Years=29]
+----
+
+
+== Map driver records to a local class
+
+The earlier examples have mapped a driver value (for example a node identified with `p`) to a class.
+You can also use the mapping feature with driver records, through the link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Record.html#as(java.lang.Class)[`Record.as()`] method.
+
+.Return keys `name`/`p.age` are mapped to the object attributes `Name`/`Age`
+[source, java]
+----
+// import org.neo4j.driver.mapping.Property;
+
+record Person(String name, @Property("p.age") Integer age) {}
+var persons = driver.executableQuery("""
+            MERGE (p:Person {name: 'Margarida', age: 29})
+            RETURN p.name AS name, p.age
+        """)
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.as(Person.class))
+    .toList();
+System.out.println(persons.get(0));  // Person[name=Margarida, age=29]
+----
+
+
+[#constructors]
+== Work with multiple constructors
+
+Your Java record class can contain multiple constructors.
+In that case, the driver picks one basing on the following criteria (in order of priority):
```

Comment:
```suggestion
In that case, the driver picks one based on the following criteria (in order of priority):
```

Reply:
If the sentence was in passive form, `based on` would be correct. Because the sentence is active, `basing` is correct.